sql/src/main/java/io/crate/metadata/table/ConstraintInfo.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.metadata.table;
+
+import io.crate.metadata.TableIdent;
+import org.apache.lucene.util.BytesRef;
+
+public class ConstraintInfo {
+
+    public enum Constraint {
```

Comment:
documentation please (CHECK?)

Reply:
will add the doku. About the CHECK:  Afaik there is no NOT NULL constraint in the sql-99 standard. Check is also what posgres is using. 

https://en.wikipedia.org/wiki/Check_constraint#NOT_NULL_constraint